# Release 3.0.0

golluroSICKAG opened the pull request:
# Version 3.0.0

## New features
- Supports FlowConfig feature to forward loaded images
- Provide version of module via 'OnNewStatusModuleVersion'
- Function 'getParameters' to provide PersistentData parameters
- Check if features of module can be used on device and provide this via 'OnNewStatusModuleIsActive' event / 'getStatusModuleActive' function
- Function to 'resetModule' to default setup
- Provide new images in /resources

## Improvements
- Provide available folders auto suggestion in UI
- New UI design available (e.g. selectable via CSK_Module_PersistentData v4.1.0 or higher), see 'OnNewStatusCSKStyle'
- 'loadParameters' returns its success
- 'sendParameters' can control if sent data should be saved directly by CSK_Module_PersistentData
- Changed log level of some messages from 'info' to 'fine'
- Added UI icon and browser tab information

## Following tasks were checked

- [x] Module is coded and structured according to the "Developing guideline for modules" described within the CSK documentation
- [x] All functions/events/parameters are documented within the manifest documentation
- [x] The manifest description of the main CROWN includes main information about the purpose of the module and how to use it in general
- [x] API docu based on the manifest was created and stored within the "docu"-folder of the repository
- [x] Internal LUA code documentation exists for variables and non served functions
- [x] All relevant infos are logged via the SharedLogger 'ModuleLogger'
- [x] Module supports persistent data feature based on 'CSK_Module_PersistentData'
- [x] Module supports user management based on 'CSK_Module_UserManagement'
- [x] No open "ToDos" within the code or at least clearly explained comments why they exist...
- [x] "Version" key in app manifest was updated following semantic versioning (and use '0.x.y' for test / experimental modules which are not yet ready to be officially released)
- [x] Meaningful IDs used for UI elements
- [x] Module was tested on an AppSpace device (at least on SICK AppEngine) with no error message
- [x] README.md is up to date (incl. info of device + firmware the module was tested with)
- [x] CHANGELOG.md is up to date

dzekomiSICKAG reviewed (approved):
Tested on AppEngine 2.0.0 on TDC-E

kogliinSICKAG commented:
1. I have seen in the FlowConfig that the ImagePlayer is a provider block but without any adjustable parameters. Is it possible to implement the parameters like cycle time into the FlowConfig?
2. The new GUI misses units for numbers. I think the units are necessary to be given at least somewhere. I suggest to change the label from "Cycle time:" to "Cycle time [ms]:".
3. Is it meant that the GUI has some differences compared to the MultiImageFilter? The viewer could have the same size in both modules, there should be enough space to put the buttons further down. Also some positions of elements at the top move in comparison with the MultiImageFilter.
4. The Image resize factor is not that obviously clear to me. Is it a percentage of the original image? Is it a factor that is multiplied on each of the two dimensions? In comparison to SICK Nova, there a downsampling factor is used from 1.00 to something like maybe 20.00.
5. An idea for the future might be interesting for DeepLearning: Have a randomized order of the pictures within the provided folder and the option to run only once over the images.

golluroSICKAG commented:
@kogliinSICKAG 
1) Would be possible, please create an issue within the repository if this feature is needed.
2) Done
3) Fixed (viewer has same size now)
4) Added some docu
5) Please create an issue within the repository.

Thx for the review!
